pkg/convert/xliff2.go

```diff
@@ -0,0 +1,94 @@
+package convert
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"go.expect.digital/translate/pkg/model"
+	"golang.org/x/text/language"
+)
+
+// XLIFF 2 Specification: https://docs.oasis-open.org/xliff/xliff-core/v2.0/os/xliff-core-v2.0-os.html
+// XLIFF 2 Example: https://localizely.com/xliff-file/?tab=xliff-20
+
+type xliff2 struct {
+	XMLName xml.Name     `xml:"urn:oasis:names:tc:xliff:document:2.0 xliff"`
+	Version string       `xml:"version,attr"`
+	SrcLang language.Tag `xml:"srcLang,attr"`
+	File    file         `xml:"file"`
+}
+type file struct {
+	Units []unit `xml:"unit"`
+}
+
+type unit struct {
+	ID     string  `xml:"id,attr"`
+	Notes  *[]note `xml:"notes>note"` // Set as pointer to avoid empty <notes></notes> when marshalling.
```

Comment:
I tried many solutions, but this is the only one that works: avoiding `<notes></notes>` when marshaling `xliff2` struct with no `Notes`.
If there is a better way/ways I am happy to hear about them :)

Reply:
try `omitempty`

```suggestion
	Notes  []note `xml:"notes>note,omitempty"`
```

Reply:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<xliff xmlns="urn:oasis:names:tc:xliff:document:2.0" version="2.0" srcLang="en">
  <file>
    <unit id="Welcome">
      <notes>
        <note category="description">To welcome a new visitor</note>
      </notes>
      <segment>
        <source>Welcome to our website!</source>
      </segment>
    </unit>
    <unit id="Error">
      <notes>
        <note category="description">To inform the user of an error</note>
      </notes>
      <segment>
        <source>Something went wrong. Please try again later.</source>
      </segment>
    </unit>
    <unit id="Feedback">
      <notes></notes>
      <segment>
        <source>We appreciate your feedback. Thank you for using our service.</source>
      </segment>
    </unit>
  </file>
</xliff>

```

Reply:
Not possible using field tag https://github.com/golang/go/issues/7233

It is possible by implementing `xml.Marshaler`.